File: problems/s/sibice/sibice.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""

from math import sqrt

def main():
    n, w, h = [int(val) for val in input().split()]

    max_length = int(sqrt(w ** 2 + h ** 2))

    for _ in range(n):
        l = int(input())

        if l <= max_length:
            print("DA")
        else:
            print("NE")


if __name__ == '__main__':
    main()

File: problems/s/sibice/sibice.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <iostream>
#include <cmath>

using namespace std;

int main() {
    int n, w, h, max_length, l;

    cin >> n >> w >> h;

    max_length = sqrt(pow(w, 2) + pow(h, 2));

    for (int i=0; i<n; i++) {
        cin >> l;

        if (l <= max_length) {
            cout << "DA" << endl;
        }
        else {
            cout << "NE" << endl;
        }
    }

    return 0;
}